File: php/spec/email.spec.php
<?php declare(strict_types=1);

use Enotype\Loaders;

describe('email', function() {
  describe('with \'john.doe@eno-lang.org\'', function() {
    it('produces the expected result', function() {
      expect(Loaders::email('john.doe@eno-lang.org'))->toEqual('john.doe@eno-lang.org');
    });
  });
  
  describe('with \'john.doe@eno-lang\'', function() {
    it('throws an exception', function() {
      expect(function() { Loaders::email('john.doe@eno-lang'); })->toThrow();
    });
  });
  
  describe('with \'@eno-lang.org\'', function() {
    it('throws an exception', function() {
      expect(function() { Loaders::email('@eno-lang.org'); })->toThrow();
    });
  });
  
  describe('with \'john.doe@.org\'', function() {
    it('throws an exception', function() {
      expect(function() { Loaders::email('john.doe@.org'); })->toThrow();
    });
  });
});

File: ruby/spec/email.spec.rb
require 'enotype'

describe 'email' do
  context 'with \'john.doe@eno-lang.org\'' do
    it 'produces the expected result' do
      expect(Enotype.email('john.doe@eno-lang.org')).to eq('john.doe@eno-lang.org')
    end
  end
  
  context 'with \'john.doe@eno-lang\'' do
    it 'raises an error' do
      expect { Enotype.email('john.doe@eno-lang') }.to raise_error(RuntimeError)
    end
  end
  
  context 'with \'@eno-lang.org\'' do
    it 'raises an error' do
      expect { Enotype.email('@eno-lang.org') }.to raise_error(RuntimeError)
    end
  end
  
  context 'with \'john.doe@.org\'' do
    it 'raises an error' do
      expect { Enotype.email('john.doe@.org') }.to raise_error(RuntimeError)
    end
  end
end